EpiAware/src/epimodel.jl

```diff
@@ -38,9 +38,43 @@ struct EpiModel{T<:Real} <: AbstractEpiModel
         #construct observation delay kernel
         K = zeros(time_horizon, time_horizon) |> SparseMatrixCSC
         for i = 1:time_horizon, j = 1:time_horizon
-            m = (i - 1) - (j - 1)
-            if m >= 1 && m <= length(delay_int)
-                K[i, j] = delay_int[m]
+            m = i - j
+            if m >= 0 && m <= (length(delay_int) - 1)
+                K[i, j] = delay_int[m+1]
+            end
+        end
+
+        new{eltype(gen_int)}(
+            gen_int,
+            delay_int,
+            K,
+            cluster_coeff,
+            length(gen_int),
+            length(delay_int),
+        )
+    end
+
+    function EpiModel(
+        gen_distribution::ContinuousDistribution,
+        delay_distribution::ContinuousDistribution,
+        cluster_coeff,
+        time_horizon;
+        Δd = 1.0,
+        D_gen,
+        D_delay,
+    )
+        gen_int =
+            create_discrete_pmf(gen_distribution, Δd = Δd, D = D_gen) |>
+            p -> p[2:end] ./ sum(p[2:end])
```

Comment:
https://github.com/CDCgov/Rt-without-renewal/blob/04897159395ee0d2c543e10b8f990c36074e4fa8/EpiAware/test/test_epimodel.jl#L92-L93

Reply:
Oh sorry I meant something like an inline assert. I don't know if that's programmatically a good idea though. I'm sure it works for _this_ GI, but I'm not sure the renormalization is enough for every GI. Floating points are a bit of dark magic to me.

Reply:
I'm not sure what the problem is here? Are you worried that `create_discrete_pmf` might have a problem or that left trunction might have a problem?

Reply:
I'm worried that after left truncating and renormalizing the `gen_int` still might not sum to one for all potential GIs. In R

```r
y <- x / sum(x)
abs(1 - sum(y)) < 1e-8
```

doesn't evaluate to `TRUE` for all `x`. I _think_ this is due to bad floating point math in R (not sure what else this would be). We had to toss the normalization into a loop to ensure that the `gen_int` vector would sum 1 to within numerical tolerance ([see here](https://github.com/cdcent/cfa-nnh-pipelines/blob/a30c421382d4f6980c6b3a3566a67077b0340a0d/NHSN/Rt/R/disease_process.R#L47)).

But if you think this isn't a concern in Julia, if defer to your expertise here.

Reply:
The $\approx$ function is [here](https://docs.julialang.org/en/v1/base/math/#Base.isapprox).

Reply:
I think this is pretty reliable.

Reply:
Good enough for me!